pandas/tests/dtypes/test_common.py

```diff
@@ -124,3 +121,403 @@ def test_get_dtype_error_catch(func):
     # No exception should be raised.
 
     assert not func(None)
+
+
+def test_is_object():
+    assert com.is_object_dtype(object)
+    assert com.is_object_dtype(np.array([], dtype=object))
+
+    assert not com.is_object_dtype(int)
+    assert not com.is_object_dtype(np.array([], dtype=int))
+    assert not com.is_object_dtype([1, 2, 3])
+
+
+def test_is_sparse():
+    assert com.is_sparse(pd.SparseArray([1, 2, 3]))
+    assert com.is_sparse(pd.SparseSeries([1, 2, 3]))
+
+    assert not com.is_sparse(np.array([1, 2, 3]))
+
+    # This test will only skip if the previous assertions
+    # pass AND scipy is not installed.
+    sparse = pytest.importorskip("scipy.sparse")
+    assert not com.is_sparse(sparse.bsr_matrix([1, 2, 3]))
+
+
+def test_is_scipy_sparse():
+    tm._skip_if_no_scipy()
+
+    from scipy.sparse import bsr_matrix
+    assert com.is_scipy_sparse(bsr_matrix([1, 2, 3]))
+
+    assert not com.is_scipy_sparse(pd.SparseArray([1, 2, 3]))
+    assert not com.is_scipy_sparse(pd.SparseSeries([1, 2, 3]))
+
+
+def test_is_categorical():
+    cat = pd.Categorical([1, 2, 3])
+    assert com.is_categorical(cat)
+    assert com.is_categorical(pd.Series(cat))
```

Comment:
I might have wanted to add this before you merged, but oh well, can do in follow-up.